backend/src/layer/data/image_validation/api_request.rs

```diff
@@ -0,0 +1,93 @@
+use crate::interface::image_validation::ImageValidationError::InvalidResponse;
+use crate::interface::image_validation::Result;
+use crate::layer::data::image_validation::json_structs::{
+    ResponseJson, SafeSearchJson, SafeSearchResponseJson,
+};
+use async_graphql::http::receive_batch_json;
+use google_jwt_auth::AuthConfig;
+use std::fs;
+
+static API_REST_URL: &str = "https://vision.googleapis.com/v1/images:annotate";
+static API_USAGE: &str = "https://www.googleapis.com/auth/cloud-vision";
+static PROJECT_ID_HEADER: &str = "x-goog-user-project";
+static REQUEST_TYPE: &str = "SAFE_SEARCH_DETECTION";
+static CONTENT_TYPE: &str = "application/json";
+static TOKEN_LIFETIME: i64 = 3600;
+static CHARSET: &str = "utf-8";
+
+//TODO DOC
+pub struct ApiRequest {
+    google_project_id: String,
+    auth_config: AuthConfig,
+}
+
+impl ApiRequest {
+    //TODO DOC
+    pub fn new(service_account_json_path: String, google_project_id: String) -> Result<Self> {
+        let json_str = fs::read_to_string(service_account_json_path)?;
+
+        Ok(Self {
+            google_project_id,
+            auth_config: AuthConfig::build(json_str, String::from(API_USAGE))?,
+        })
+    }
+
+    //TODO DOC
+    pub async fn encoded_image_validation(&self, b64_image: String) -> Result<SafeSearchJson> {
+        let token = self.auth_config.generate_auth_token(TOKEN_LIFETIME).await?;
+        let json_resp = self.request_api(b64_image, token).await?.responses.pop();
+        match json_resp {
+            None => Err(InvalidResponse),
+            Some(json) => Ok(json.safeSearchAnnotation),
+        }
+    }
+
+    async fn request_api(
+        &self,
+        b64_image: String,
+        auth_token: String,
+    ) -> Result<SafeSearchResponseJson> {
+        let resp = reqwest::Client::new()
+            .post(API_REST_URL)
+            .header(
+                reqwest::header::AUTHORIZATION,
+                format!("Bearer {auth_token}"),
+            )
+            .header(PROJECT_ID_HEADER, &self.google_project_id)
+            .header(reqwest::header::CONTENT_TYPE, CONTENT_TYPE)
+            .header(reqwest::header::ACCEPT_CHARSET, CHARSET)
+            .body(build_request_body(&b64_image))
+            .send()
+            .await?;
+        // TODO retry with error json if response could not be decoded.
+        // TODO For now, this decode error (containing the response error json)..
+        // TODO ..will be displayed as decode error and not as api error.
+        Ok(resp.json::<SafeSearchResponseJson>().await?)
+    }
+}
+
+fn build_request_body(b64_image: &str) -> String {
+    format!(
+        r#"{{"requests":[{{"image":{{"content":"{b64_image}"}},"features":[{{"type":"{REQUEST_TYPE}"}},]}}]}}"#
```

Comment:
maybe struct with serde?

Reply:
btw this is not an formatting string, `b64_image` will NOT be inserted

Reply:
but it works afaik

Reply:
id somehow did not read oorrectly...